plugins/modules/katello_content_view.py

```diff
@@ -95,106 +98,109 @@
 
     # only update CVC's of newly created or updated CCV's
-    if state == 'present' or (state == 'present_with_defaults' and changed):
-        current_cvcs = []
-        if hasattr(content_view_entity, 'content_view_component'):
-            current_cvcs = [cvc.read() for cvc in content_view_entity.content_view_component]
-        if 'components' in entity_dict and content_view_entity.composite:
-            for component in entity_dict['components']:
-                cvc = component.copy()
-                cvc['content_view'] = find_content_view(module, name=component['content_view'], organization=entity_dict['organization'])
-                cvc_matched = None
-                for _cvc in current_cvcs:
-                    if _cvc.content_view.id == cvc['content_view'].id:
-                        cvc_matched = _cvc
-                force_update = list()
-                if 'version' in component:
-                    cvc['version'] = find_content_view_version(module, cvc['content_view'], version=component['version'])
+    # Also only act for composite content views
+    if (module.state == 'present' or (module.state == 'present_with_defaults' and changed)) and content_view_entity['composite'] and components is not None:
+        if not changed:
+            content_view_entity['content_view_components'] = entity['content_view_components']
+        current_cvcs = content_view_entity.get('content_view_components', [])
+        if components:
+            components_to_add = []
+            ccv_scope = {'composite_content_view_id': content_view_entity['id']}
+            for component in components:
+                cvc = {
+                    'content_view': module.find_resource_by_name('content_views', name=component['content_view'], params=scope),
+                    'latest': component['latest'],
+                }
+                cvc_matched = next((item for item in current_cvcs if item['content_view']['id'] == cvc['content_view']['id']), None)
+                if not cvc['latest']:
+                    version = component.get('content_view_version')
+                    if version is None:
+                        module.fail_json(msg="Content View Component must either have latest=True or provide a Content View Version.")
+                    search = "content_view_id={},version={}".format(cvc['content_view']['id'], component['content_view_version'])
+                    cvc['content_view_version'] = module.find_resource('content_view_versions', search=search, thin=True)
                     cvc['latest'] = False
-                    if cvc_matched and cvc_matched.latest:
+                    if cvc_matched and cvc_matched['latest']:
                         # When changing to latest=False & version is the latest we must send 'content_view_version' to the server
-                        force_update.append('content_view_version')
+                        # Let's fake, it wasn't there...
```

Comment:
not sure what you're faking by removing things

Reply:
or do you fake it by removing stuff here, so that `ensure_entity` will add it to the fields to send?

Reply:
exactly. This was called `force_update` before, i think.